### PR TITLE
feat: 756 postgre sql to big query

### PR DESCRIPTION
**Pull request by gabrielacorona**
postgresql to bigquery notebook with parameterization script

### CODE REVIEW

File: notebooks/postgresql2bq/PostgreSqlToBigQuery_parameterize_script.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from os import environ
+from typing import Dict, Sequence, Optional, Any
+import argparse
+import json
```
Comment (tanyarw) [inline notebooks/postgresql2bq/PostgreSqlToBigQuery_parameterize_script.py:18]: Unused import can be removed

File: notebooks/postgresql2bq/PostgreSqlToBigQuery_parameterize_script.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            'Leave empty for migrating complete database else provide tables as \"table1,table2\"'
+        )
+
+        parser.add_argument(
+            f'--{constants.POSTGRESQL_SCHEMA_LIST_ARG}',
+            dest=constants.POSTGRESQL_SCHEMA_LIST,
+            required=False,
+            default='',
+            help='POSTGRESQL schema list to migrate. '
+            'Only Migrate tables associated with the provided schema list'
```
Comment (tanyarw) [inline notebooks/postgresql2bq/PostgreSqlToBigQuery_parameterize_script.py:97]: Can we mention to provide the schema list as 'schema1,schema2'

File: notebooks/postgresql2bq/PostgreSqlToBigQuery_parameterize_script.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+    def run(self, args: Dict[str, Any]) -> None:
+        """
+        Run the notebook.
+        """
+
+        # Convert comma separated string to list
+        args[constants.POSTGRESQL_TABLE_LIST] = list(
+            map(str.strip, args[constants.POSTGRESQL_TABLE_LIST].split(","))
+        )
```
Comment (tanyarw) [inline notebooks/postgresql2bq/PostgreSqlToBigQuery_parameterize_script.py:155]: Looks like `args[constants.POSTGRESQL_SCHEMA_LIST]` must also be converted from a string to list, similar to what has been done for `POSTGRESQL_TABLE_LIST`

File: notebooks/postgresql2bq/PostgreSqlToBigQuery_parameterize_script.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+        parser.add_argument(
+            f'--{constants.MAX_PARALLELISM_ARG}',
+            dest=constants.MAX_PARALLELISM,
+            type=int,
+            default=5,
+            required=False,
+            help='Maximum number of tables that will migrated parallelly (Default: 5)'
+        )
+
```
Comment (tanyarw) [inline notebooks/postgresql2bq/PostgreSqlToBigQuery_parameterize_script.py:124]:
```suggestion
parser = get_common_args(parser)

```

File: notebooks/postgresql2bq/PostgreSqlToBigQuery_parameterize_script.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+from os import environ
+from typing import Dict, Sequence, Optional, Any
+import argparse
+import json
+
+import papermill as pm
+
+from parameterize_script import BaseParameterizeScript
+import parameterize_script.util.notebook_constants as constants
+
```
Comment (tanyarw) [inline notebooks/postgresql2bq/PostgreSqlToBigQuery_parameterize_script.py:24]:
```suggestion
from parameterize_script.util import get_common_args

```

**Review by tanyarw — CHANGES_REQUESTED** [review]
Hi Gaby, 
Once you pull the latest changes & resolve merge conflicts, please apply these suggested changes and test the job
Thanks!

File: notebooks/postgresql2bq/PostgreSqlToBigQuery_parameterize_script.py [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+        parser = argparse.ArgumentParser()
+
+        parser.add_argument(
+            f'--{constants.OUTPUT_NOTEBOOK_ARG}',
+            dest=constants.OUTPUT_NOTEBOOK_ARG,
+            required=False,
+            default=None,
+            help='Path to save executed notebook (Default: None). '
+            'If not provided, no notebook is saved'
+        )
```
Comment (tanyarw) [inline notebooks/postgresql2bq/PostgreSqlToBigQuery_parameterize_script.py:44]:
```suggestion
```

File: notebooks/postgresql2bq/PostgreSqlToBigQuery_parameterize_script.py [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+        """
+        Run the notebook.
+        """
+
+        # Convert comma separated string to list
+        args[constants.POSTGRESQL_TABLE_LIST] = list(
+            map(str.strip, args[constants.POSTGRESQL_TABLE_LIST].split(","))
+        )
+        # Exclude arguments that are not needed to be passed to the notebook
+        ignore_keys = {constants.OUTPUT_NOTEBOOK_ARG}
```
Comment (tanyarw) [inline notebooks/postgresql2bq/PostgreSqlToBigQuery_parameterize_script.py:157]:
```suggestion
        ignore_keys = {constants.LOG_LEVEL_ARG, constants.OUTPUT_NOTEBOOK_ARG}
```

File: notebooks/postgresql2bq/postgresql-to-bigquery-notebook.ipynb [hunk 8]
```diff
@@ -0,0 +1,2413 @@
...
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Service Account:  617357862702-compute@developer.gserviceaccount.com\n"
+     ]
+    }
+   ],
+   "source": [
+    "SERVICE_ACCOUNT = \"\" \n",
```
Comment (tanyarw) [inline notebooks/postgresql2bq/postgresql-to-bigquery-notebook.ipynb:303]: `SERVICE_ACCOUNT` has already been initialized in the previous cell

File: notebooks/postgresql2bq/postgresql-to-bigquery-notebook.ipynb [hunk 9]
```diff
@@ -0,0 +1,2413 @@
...
+    "\n",
+    "POSTGRES_USERNAME : PostgreSQL username\n",
+    "\n",
+    "POSTGRES_PASSWORD : PostgreSQL password\n",
+    "\n",
+    "POSTGRES_DATABASE : name of database that you want to migrate\n",
+    "\n",
+    "OUTPUT_MODE : Output write mode (one of: append,overwrite,ignore,errorifexists)(Defaults to overwrite)\n",
+    "\n",
+    "BATCH_SIZE : JDBC output batch size. Default set to 1000"
```
Comment (tanyarw) [inline notebooks/postgresql2bq/postgresql-to-bigquery-notebook.ipynb:357]: nit: Markdown description doesn't match variable names 😬

File: notebooks/postgresql2bq/postgresql-to-bigquery-notebook.ipynb [hunk 10]
```diff
@@ -0,0 +1,2413 @@
...
+   "source": [
+    "POSTGRESQL_HOST=\"10.0.0.11\"\n",
+    "POSTGRESQL_PORT=\"5432\"\n",
+    "POSTGRESQL_USERNAME=\"my_username\"\n",
+    "POSTGRESQL_PASSWORD=\"my_password\"\n",
+    "POSTGRESQL_DATABASE=\"my_database\"\n",
+    "JDBCTOJDBC_OUTPUT_MODE=\"overwrite\" # one of append/overwrite/ignore/errorifexists\n",
+    "JDBCTOJDBC_OUTPUT_BATCH_SIZE=\"1000\"\n",
+    "POSTGRESQL_TABLE_LIST=[] # leave list empty for migrating complete database else provide tables as ['dbo.table1','sys.table2']. If this parameter is not empty, leave POSTGRESQL_SCHEMA_LIST empty.\n",
+    "POSTGRESQL_SCHEMA_LIST=['public'] # leave list empty for migrating complete database else provide schema as ['schema1','schema2'] for migrating all tables in specific schemas. If this parameter is not empty, leave POSTGRESQL_TABLE_LIST empty."
```
Comment (tanyarw) [inline notebooks/postgresql2bq/postgresql-to-bigquery-notebook.ipynb:379]: This looks like a repeat of variables declared in the previous code cell, right?

**Review by tanyarw — COMMENTED** [review]
Hey Gaby, a few more comments

**Review by tanyarw — APPROVED** [review]

File: notebooks/postgresql2bq/postgresql-to-bigquery-notebook.ipynb (hunk 8 above)
Comment (gabrielacorona) [reply notebooks/postgresql2bq/postgresql-to-bigquery-notebook.ipynb:303]: i think this is part of the metadata of when the notebook gets run

File: notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb [hunk 11]
```diff
@@ -387,7 +387,7 @@
    "outputs": [],
    "source": [
     "pkDF = pd.DataFrame({\"table\" : SQL_SERVER_TABLE_LIST, \"primary_keys\": list(SQL_TABLE_PRIMARY_KEYS.values())})\n",
-    "print(\"Below are identified primary keys for migrating mssql table to postgres:\")\n",
+    "print(\"Below are identified primary keys for migrating mssql table to bigquery:\")\n",
```
Comment (NiloFreitas) [inline notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb:390]: postgresql instead of mssql

**Review by NiloFreitas — COMMENTED** [review]
LGTM.
Great work.
Please address the last comment and add the PostgreSQLToBigQuery bullet point at the bottom of the notebooks README file

**Pull request merged**